optimizer_wrapper.rb

```diff
@@ -501,6 +504,39 @@ def self.job_remove(api_key, id)
 
   def self.check_result_consistency(expected_value, results)
     [results].flatten(1).each{ |result|
+      if result[:routes].any?{ |route| route[:activities].any?{ |a| a[:waiting_time].to_i < 0 } }
+        log 'Computed waiting times are invalid', level: :warn
+        raise RuntimeError, 'Computed waiting times are invalid' if ENV['APP_ENV'] != 'production'
+      end
+
+      waiting_times = result[:routes].map{ |route| route[:total_waiting_time] }.compact
+      durations = result[:routes].map{ |route|
+        route[:activities].map{ |act| act[:departure_time] && (act[:departure_time] - act[:begin_time]) }.compact
+      }
+      previous_point = nil
+      setup_durations = result[:routes].map{ |route|
+        route[:activities].map{ |act|
+          next if act[:type] == 'rest'
+
+          setup = (act[:travel_time].nil? || act[:travel_time]&.positive? ? act[:detail][:setup_duration] : 0)
+          previous_point = act[:point_id]
+          setup
+        }.compact
+      }
+      total_time = result[:total_time] || 0
+      total_travel_time = result[:total_travel_time] || 0
+      if total_time != (total_travel_time || 0) +
+                       waiting_times.sum +
+                       (setup_durations.flatten.reduce(&:+) || 0) +
+                       (durations.flatten.reduce(&:+) || 0)
+
+        log_string = "Expected #{total_time} == #{total_travel_time} +"\
+                     " #{waiting_times.sum} + #{setup_durations.flatten.reduce(&:+)}"\
+                     " + #{durations.flatten.reduce(&:+)}"
+        log log_string, level: :warn
+        raise RuntimeError, 'Computed times are invalid' if ENV['APP_ENV'] != 'production'
```

Comment:
Same here, could we use warn instead of raise ? We could return none of these errored fields but still return a result to customer ?

Reply:
We do log a warn, when in production environment, and raise the error while in test or in development environments ! :)